File: app/src/main/java/com/druger/aboutwork/presenters/BasePresenter.java
package com.druger.aboutwork.presenters;

import android.support.annotation.NonNull;

import com.arellomobile.mvp.MvpPresenter;
import com.arellomobile.mvp.MvpView;
import com.druger.aboutwork.db.RealmHelper;
import com.druger.aboutwork.rest.RestApi;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;
import timber.log.Timber;

/**
 * Created by druger on 20.07.2017.
 */

public class BasePresenter<View extends MvpView> extends MvpPresenter<View> {

    protected String TAG = getClass().getSimpleName();

    protected RestApi restApi;
    protected RealmHelper realmHelper;

    private CompositeDisposable compositeDisposable = new CompositeDisposable();

    protected final void unSubscribeOnDestroy(@NonNull Disposable disposable) {
        compositeDisposable.add(disposable);
    }

    protected void handleError(Throwable throwable) {
        Timber.e(throwable);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        compositeDisposable.clear();
    }
}

File: app/src/main/java/com/druger/aboutwork/presenters/BasePresenter.kt
package com.druger.aboutwork.presenters

import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable
import moxy.MvpPresenter
import moxy.MvpView
import timber.log.Timber

/**
 * Created by druger on 20.07.2017.
 */

open class BasePresenter<View : MvpView> : MvpPresenter<View>() {

    private val compositeDisposable = CompositeDisposable()

    protected fun unSubscribeOnDestroy(disposable: Disposable) {
        compositeDisposable.add(disposable)
    }

    protected open fun handleError(throwable: Throwable) {
        Timber.e(throwable)
    }

    override fun onDestroy() {
        super.onDestroy()
        compositeDisposable.clear()
    }
}
